test/integration/statefulset/statefulset_test.go

```diff
@@ -489,6 +489,137 @@ func TestAutodeleteOwnerRefs(t *testing.T) {
 	}
 }
 
+func TestDeletingPodForRollingUpdatePartition(t *testing.T) {
```

Comment:
Maybe we could drop it given the e2e test can reproduce it consistently. WDYT @soltysh ?

Reply:
Updated the test according to your PR: https://github.com/kubernetes/kubernetes/pull/120921 , now integration should always fail.

Reply:
Please resolve, I support going with e2e + integration. 
- point for e2e: it demonstrates the issue with kubelet, which was changed in 1.27+ that every pod transitions to terminal phase before deletion.
- point for integration: it demonstates the issue with handling of Failed phase, which is very tricky to reproduce before 1.27. It requires eviction, for example due to node pressure.